conda_build/build.py

```diff
@@ -2760,7 +2760,15 @@ def build(
                     #    can be different from the env for the top level build.
                     with utils.path_prepended(m.config.build_prefix):
                         env = environ.get_dict(m=m)
-                    pkg_type = "conda" if not hasattr(m, "type") else m.type
+
+                    if not hasattr(m, "type"):
+                        if m.config.conda_pkg_format == CondaPkgFormat.V2:
+                            pkg_type = CondaPkgFormat.V2
+                        else:
+                            pkg_type = CondaPkgFormat.V1
+                    else:
+                        pkg_type = m.type
+
```

Comment:
Should it just be:

```python
pkg_type = getattr(m, "type", m.config.conda_pkg_format)
```

?

Reply:
Yes `pkg_type = getattr(m, "type", m.config.conda_pkg_format)` would also work assuming `m.config.conda_pkg_format` is defined and not `None`. The PR i wrote additionally handles that case by defaulting to `CondaPkgFormat.V1`.

Reply:
Then we can add: 

```python
pkg_type = getattr(m, "type", m.config.conda_pkg_format) or CondaPkgFormat.V2
```

Note we default to V2 now (will be released in 25.1).

Reply:
Sounds good!

Reply:
```suggestion
                    pkg_type = getattr(m, "type", m.config.conda_pkg_format) or CondaPkgFormat.V2
```